### PR TITLE
[MU4] Parts dialog

### PR DESCRIPTION
**Pull request by RomanPudashkin**
The basis for new parts dialog

 _(this is not final design)_

### CODE REVIEW

**Review by igorkorsukov — APPROVED** [review]

File: mu4/notation/qml/MuseScore/NotationScene/PartsDialog.qml [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        ItemSelectionModel {
+            id: selectionModel
+            model: partsModel
+        }
+
+        Component.onCompleted: {
+            partsModel.load()
+        }
+
+        Column {
```
Comment (Eism) [inline mu4/notation/qml/MuseScore/NotationScene/PartsDialog.qml:35]: What happens if the user decides to stretch the window?
I think it's better to use ColumnLayout

File: mu4/notation/qml/MuseScore/NotationScene/PartsDialog.qml [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                        anchors.leftMargin: view.margin
+
+                        text: model.title
+
+                        font.bold: true
+                    }
+
+                    SeparatorLine { anchors.bottom: parent.bottom }
+
+                    property bool selected: false
```
Comment (Eism) [inline mu4/notation/qml/MuseScore/NotationScene/PartsDialog.qml:143]: Can you just set 
property bool selected: selectionModel.hasSelection ?

File: mu4/notation/view/notationswitchlistmodel.h [hunk 3]
```diff
@@ -33,7 +33,7 @@ class NotationSwitchListModel : public QAbstractListModel, public async::Asyncab
 {
     Q_OBJECT
 
-    INJECT(notation, context::IGlobalContext, globalContext)
+    INJECT(notation, context::IGlobalContext, context)
```
Comment (Eism) [inline mu4/notation/view/notationswitchlistmodel.h:36]: why renamed? :)

File: mu4/notation/qml/MuseScore/NotationScene/PartsDialog.qml (hunk 1 above)
Comment (RomanPudashkin) [reply mu4/notation/qml/MuseScore/NotationScene/PartsDialog.qml]: fixed

File: mu4/notation/qml/MuseScore/NotationScene/PartsDialog.qml (hunk 2 above)
Comment (RomanPudashkin) [reply mu4/notation/qml/MuseScore/NotationScene/PartsDialog.qml:142]: No, I can't. This is the property of a delegate. hasSelection indicated that any item is selected

File: mu4/notation/view/notationswitchlistmodel.h (hunk 3 above)
Comment (RomanPudashkin) [reply mu4/notation/view/notationswitchlistmodel.h:36]: We use this version in many places. I think that is more simple

**Pull request merged**